addons/website_event_exhibitor/controllers/exhibitor.py

```diff
@@ -77,16 +78,22 @@ def _event_exhibitors_get_values(self, event, **searches):
         sponsor_types = sponsors_all.mapped('sponsor_type_id')
         sponsor_countries = sponsors_all.mapped('partner_id.country_id').sorted('name')
         # organize sponsors into categories to help display
-        sponsor_categories = dict()
+        sponsor_categories_dict = defaultdict(request.env['event.sponsor'].sudo)
+        sponsor_categories = []
+        is_event_user = request.env.user.has_group('event.group_event_user')
         for sponsor in sponsors:
-            if not sponsor_categories.get(sponsor.sponsor_type_id):
-                sponsor_categories[sponsor.sponsor_type_id] = request.env['event.sponsor'].sudo()
-            sponsor_categories[sponsor.sponsor_type_id] |= sponsor
-        sponsor_categories = [
-            dict({
+            sponsor_categories_dict[sponsor.sponsor_type_id] |= sponsor
+
+        for sponsor_category, sponsors in sponsor_categories_dict.items():
+            # To display random published sponsors first and random unpublished sponsors last
```

Comment:
Does not seem random anymore, except the "or" that does not guarantee order ?

Reply:
Right.. Logic for random sponsor was implemented (see commit from 2nd April - https://bit.ly/3gewVya), but was lost at some point when I was on leave last month. It's re-introduced now :)